src/platform/nxp/k32w/k32w0/OTAImageProcessorImpl.cpp

```diff
@@ -142,29 +142,30 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
 
 CHIP_ERROR OTAImageProcessorImpl::SetBlock(ByteSpan & block)
 {
-    if ((block.data() == nullptr) || block.empty())
+    if (!IsSpanUsable(block))
     {
+        ReleaseBlock();
```

Comment:
Why this release?

Reply:
Agreed, there would be nothing to release

Reply:
agree, removed the unneeded release.